src/server/status_server.rs

```diff
@@ -451,6 +452,40 @@ impl StatusServer {
             acceptor.set_ca_file(&security_config.ca_path)?;
             acceptor.set_certificate_chain_file(&security_config.cert_path)?;
             acceptor.set_private_key_file(&security_config.key_path, SslFiletype::PEM)?;
+
+            if !security_config.cert_allowed_cn.is_empty() {
+                let allowed_cn = security_config.cert_allowed_cn.clone();
+                // The verification callback to check if the peer CN is allowed.
+                let verify_cb = move |flag: bool, x509_ctx: &mut X509StoreContextRef| {
+                    if !flag || x509_ctx.error_depth() != 0 {
+                        return flag;
+                    }
+                    if let Some(chains) = x509_ctx.chain() {
+                        if chains.len() != 0 {
+                            if let Some(pattern) = chains
+                                .get(0)
+                                .unwrap()
+                                .subject_name()
+                                .entries_by_nid(openssl::nid::Nid::COMMONNAME)
+                                .next()
```

Comment:
Could you iter chains and subject_names?

Reply:
@tennix PTAL

Reply:
Just need to verify the first cert

Reply:
Can a cert contain multiple common names?

Reply:
No, so just next once

Reply:
No, Just `next()` once